planet/src/main/webapp/places.js

```diff
@@ -145,29 +158,39 @@ function setSearchByButton() {
 
 /** Create markers and add details to each place */
 function addPlaceDetails() {
-    // For each place, get the icon, name and location.
-    places.forEach(function(place) {
-        if (!place.geometry) {
-          console.log("Returned place contains no geometry");
-          return;
-        }
-        createMarkers(place); 
-        if (place.geometry.viewport) {
-            bounds.union(place.geometry.viewport);
-        } else {
-            bounds.extend(place.geometry.location);
-        }
+    const half = Math.ceil(places.length /2); 
+    const firstHalf = places.splice(0,half);
+    const secondHalf = places.splice(-half);
```

Comment:
if the number of results returned changes (e.g. if the maps API changes its logic), this will break.

We should write code such that it's resilient to changes that we don't control.